pywt/src/c_wt.pxd

```diff
@@ -108,25 +115,29 @@ cdef extern from "wt.h":
                           double output[], index_t output_len, int level)
 
 
-    cdef int float_dec_a(float input[], index_t input_len, Wavelet* wavelet,
-                         float output[], index_t output_len, MODE mode)
-    cdef int float_dec_d(float input[], index_t input_len, Wavelet* wavelet,
-                         float output[], index_t output_len, MODE mode)
-
-    cdef int float_rec_a(float coeffs_a[], index_t coeffs_len, Wavelet* wavelet,
-                         float output[], index_t output_len)
-    cdef int float_rec_d(float coeffs_d[], index_t coeffs_len, Wavelet* wavelet,
-                         float output[], index_t output_len)
-
-    cdef int float_idwt(float coeffs_a[], index_t coeffs_a_len, 
-                        float coeffs_d[], index_t coeffs_d_len,
-                        Wavelet* wavelet, float output[], index_t output_len,
-                        MODE mode, int correct_size)
+    cdef int float_dec_a(const float  * const input, const size_t input_len,
+                         const Wavelet * const wavelet,
+                         float  * const output, const size_t output_len,
+                         const MODE mode)
+    cdef int float_dec_d(const float  * const input, const size_t input_len,
+                         const Wavelet * const wavelet,
+                         float  * const output, const size_t output_len,
+                         const MODE mode)
+
+    cdef int float_rec_a(const float * const coeffs_a, const size_t coeffs_len,
+                         const Wavelet * const wavelet,
+                         float  * const output, const size_t output_len)
+    cdef int float_rec_d(const float * const coeffs_d, const size_t coeffs_len,
+                         const Wavelet * const wavelet,
+                         float  * const output, const size_t output_len)
+
+    cdef int float_idwt(const float * const coeffs_a, const size_t coeffs_a_len,
+                        const float * const coeffs_d, const size_t coeffs_d_len,
+                        const Wavelet * const wavelet,
+                        float  * const output, const size_t output_len,
+                        const MODE mode, const int correct_size)
 
     cdef int float_swt_a(float input[], index_t input_len, Wavelet* wavelet,
                          float output[], index_t output_len, int level)
     cdef int float_swt_d(float input[], index_t input_len, Wavelet* wavelet,
                          float output[], index_t output_len, int level)
```

Comment:
I have work in a branch to implement proper stationary wavelet transform support, where I have made this change.

`index_t` is also used in `Wavelet.support_width`, which may be negative - I assume these wavelets do not have compact support? I know the `dmey` wavelet does not, but not sure about `bior`/`rbio`